Assets/MixedRealityToolkit.Services/InputSimulation/KeyBinding.cs

```diff
@@ -0,0 +1,255 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Microsoft.MixedReality.Toolkit.Services.InputSimulation.Editor")]
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Identifier of a key combination or mouse button for generic input binding.
+    /// </summary>
+    /// <remarks>
+    /// This encodes either a KeyCode with optional modifiers or a mouse button index.
+    /// </remarks>
+    [System.Serializable]
+    public struct KeyBinding
+    {
+        /// <summary>
+        /// The type of value encoded in the <see cref="code"/> property.
+        /// </summary>
+        public enum KeyType : int
+        {
+            None = 0,
+            Mouse = 1,
+            Key = 2,
+        }
+
+        /// <summary>
+        /// Enum for interpreting the mouse button integer index.
+        /// </summary>
+        public enum MouseButton : int
+        {
+            Left = 0,
+            Right = 1,
+            Middle = 2,
+            Button3 = 3,
+            Button4 = 4,
+            Button5 = 5,
+            Button6 = 6,
+            Button7 = 7,
+        }
+
+        // Array of names to use for a combined enum selection.
+        internal static readonly string[] AllCodeNames;
+        // Maps (KeyType, code) combination onto the contiguous index used for enums.
+        // Value can be used as index in AllCodeNames array.
+        internal static readonly Dictionary<Tuple<KeyType, int>, int> KeyBindingToEnumMap;
+        // Maps enum index to a KeyBinding, for assignment after selecting an enum value.
+        internal static readonly Dictionary<int, KeyBinding> EnumToKeyBindingMap;
+
+        // Static constructor to initialize static fields
+        static KeyBinding()
+        {
+            KeyCode[] KeyCodeValues = (KeyCode[])Enum.GetValues(typeof(KeyCode));
+            string[] KeyCodeNames = Enum.GetNames(typeof(KeyCode));
```

Comment:
>KeyCodeNames [](start = 21, length = 12)

This isn't used in this constructor - was this supposed to be used? (Or can we delete this?)

Reply:
Left over from previous iteration, will remove.